api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsBatchUpdateService.java

```diff
@@ -57,24 +63,27 @@ public InitialCreditsBatchUpdateService(
   public void checkInitialCreditsUsage(List<Long> userIdList) {
     Set<String> googleProjectsForUserSet = workspaceDao.getGoogleProjectForUserList(userIdList);
 
-    List<DbGoogleProjectPerCost> googleProjectPerCostList =
-        (List<DbGoogleProjectPerCost>)
-            googleProjectPerCostDao.findAllByGoogleProjectId(googleProjectsForUserSet);
-
     // Create Map Key: googleProject and value: cost
+    Stopwatch stopwatch = stopwatchProvider.get().start();
     Map<String, Double> userWorkspaceBQCosts =
-        googleProjectPerCostList.stream()
+        getAllWorkspaceCostsFromBQ().entrySet().stream()
```

Comment:
I tried a version which added `IN(googleProjects)` to the query, but it didn't make a significant difference.  Here we do the simple thing of "just get all"